extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts

```diff
@@ -0,0 +1,506 @@
+	private searchBox?: azdata.InputBoxComponent;
+	private parametersTable!: azdata.DeclarativeTableComponent;
+	private parameterContainer?: azdata.DivContainer;
+	private _parametersTableLoading!: azdata.LoadingComponent;
+
+	private discardButton?: azdata.ButtonComponent;
+	private saveButton?: azdata.ButtonComponent;
+	private resetButton?: azdata.ButtonComponent;
+	private connectToServerButton?: azdata.ButtonComponent;
+
+	private engineSettings = `'`;
+	private engineSettingUpdates?: Map<string, string>;
+
+	private readonly _azdataApi: azdataExt.IExtension;
+
+	constructor(protected modelView: azdata.ModelView, private _postgresModel: PostgresModel) {
+		super(modelView);
+		this._azdataApi = vscode.extensions.getExtension(azdataExt.extension.name)?.exports;
+
+		this.initializeConnectButton();
+		this.initializeSearchBox();
+
+		this.engineSettingUpdates = new Map();
+
+		this.disposables.push(
+			this._postgresModel.onConfigUpdated(() => this.eventuallyRunOnInitialized(() => this.handleServiceUpdated())),
+			this._postgresModel.onEngineSettingsUpdated(() => this.eventuallyRunOnInitialized(() => this.handleEngineSettingsUpdated())));
+	}
+
+	protected get title(): string {
+		return loc.nodeParameters;
+	}
+
+	protected get id(): string {
+		return 'postgres-node-parameters';
+	}
+
+	protected get icon(): { dark: string; light: string; } {
+		return IconPathHelper.nodeParameters;
+	}
+
+	protected get container(): azdata.Component {
+		const root = this.modelView.modelBuilder.divContainer().component();
+		const content = this.modelView.modelBuilder.divContainer().component();
+		root.addItem(content, { CSSStyles: { 'margin': '20px' } });
+
+		content.addItem(this.modelView.modelBuilder.text().withProps({
+			value: loc.nodeParameters,
+			CSSStyles: { ...cssStyles.title }
+		}).component());
+
+		const info = this.modelView.modelBuilder.text().withProps({
+			value: loc.nodeParametersDescription,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const link = this.modelView.modelBuilder.hyperlink().withProps({
+			label: loc.learnAboutNodeParameters,
+			url: 'https://docs.microsoft.com/azure/azure-arc/data/configure-server-parameters-postgresql-hyperscale',
+		}).component();
+
+		const infoAndLink = this.modelView.modelBuilder.flexContainer().withLayout({ flexWrap: 'wrap' }).component();
+		infoAndLink.addItem(info, { CSSStyles: { 'margin-right': '5px' } });
+		infoAndLink.addItem(link);
+		content.addItem(infoAndLink, { CSSStyles: { 'margin-bottom': '20px' } });
+
+		content.addItem(this.searchBox!, { CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px', 'margin-bottom': '20px' } });
+
+		this.parametersTable = this.modelView.modelBuilder.declarativeTable().withProps({
+			width: '100%',
+			columns: [
+				{
+					displayName: 'Parameter Name',
+					valueType: azdata.DeclarativeDataType.component,
```

Comment:
Can the name and description columns be strings rather than components? i.e. `azdata.DeclarativeDataType.string`